### ISSUE TITLE
Cannot Mock Android's View class with mockito-android 2.8.9

### ISSUE THREAD

**Issue by bparent** [mockito/mockito#1064]
When using mockito and mockito-android 2.8.9  I cannot mock the View class from Android.

This is not an issue using dexmaker and mockito 1.10. This seems to work with dexmaker 2.2.0 (which uses mockito 2.2.0) as well.

Repro:
1. Create a new project using Android Studio
2. Add mockito and mockito-android to app/build.gradle
    ``` gradle 
      testCompile 'org.mockito:mockito-core:2.8.9'
      androidTestCompile 'org.mockito:mockito-android:2.8.9'
    ```
3. Try to mock `View.class` in `ExampleInstrumentedTest` example test
    `View view = Mockito.mock(View.class);`
4. Run the test which will fail with the following stacktrace.

Stacktrace
```java
org.objenesis.ObjenesisException: java.lang.reflect.InvocationTargetException
at org.objenesis.instantiator.android.Android18Instantiator.newInstance(Android18Instantiator.java:50)
at org.objenesis.ObjenesisBase.newInstance(ObjenesisBase.java:73)
at org.mockito.internal.creation.instance.ObjenesisInstantiator.newInstance(ObjenesisInstantiator.java:18)
at org.mockito.internal.creation.bytebuddy.SubclassByteBuddyMockMaker.createMock(SubclassByteBuddyMockMaker.java:47)
at org.mockito.android.internal.creation.AndroidByteBuddyMockMaker.createMock(AndroidByteBuddyMockMaker.java:35)
at org.mockito.internal.util.MockUtil.createMock(MockUtil.java:35)
at org.mockito.internal.MockitoCore.mock(MockitoCore.java:63)
at org.mockito.Mockito.mock(Mockito.java:1729)
at org.mockito.Mockito.mock(Mockito.java:1642)
at examplecom.mockitoviewmockfail.ExampleInstrumentedTest.useAppContext(ExampleInstrumentedTest.java:23)
at java.lang.reflect.Method.invoke(Native Method)
at java.lang.reflect.Method.invoke(Method.java:372)
at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)
at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47)
at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:325)
at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:78)
at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:57)
at org.junit.runners.ParentRunner$3.run(ParentRunner.java:290)
at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:71)
at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:288)
at org.junit.runners.ParentRunner.access$000(ParentRunner.java:58)
at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:268)
at org.junit.runners.ParentRunner.run(ParentRunner.java:363)
at org.junit.runners.Suite.runChild(Suite.java:128)
at org.junit.runners.Suite.runChild(Suite.java:27)
at org.junit.runners.ParentRunner$3.run(ParentRunner.java:290)
at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:71)
at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:288)
at org.junit.runners.ParentRunner.access$000(ParentRunner.java:58)
at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:268)
at org.junit.runners.ParentRunner.run(ParentRunner.java:363)
at org.junit.runner.JUnitCore.run(JUnitCore.java:137)
at org.junit.runner.JUnitCore.run(JUnitCore.java:115)
at android.support.test.internal.runner.TestExecutor.execute(TestExecutor.java:59)
at android.support.test.runner.AndroidJUnitRunner.onStart(AndroidJUnitRunner.java:262)
at android.app.Instrumentation$InstrumentationThread.run(Instrumentation.java:1837)
Caused by: java.lang.reflect.InvocationTargetException
at java.lang.reflect.Method.invoke(Native Method)
at java.lang.reflect.Method.invoke(Method.java:372)
at org.objenesis.instantiator.android.Android18Instantiator.newInstance(Android18Instantiator.java:47)
... 37 more
Caused by: java.lang.IllegalAccessError: Illegal class access: 'android.view.View$MockitoMock$1511770549' attempting to access 'android.view.View$AttachInfo' (declaration of 'android.view.View$MockitoMock$1511770549' appears in /data/data/examplecom.mockitoviewmockfail/cache/lmMEiAU3.jar)
at android.view.View$MockitoMock$1511770549.<clinit>(Unknown Source)
at java.io.ObjectStreamClass.newInstance(Native Method)
... 40 more
```

**Comment by raphw** [comment]
Seems like a package-private class is referenced that would only be accessible if Mockito could inject the class into the target class loader what is not always possible on Android. I will look into how to solve this better. Thanks for reporting!

**Comment by bparent** [comment]
Thanks - sorry I thought I had put in the report that AttachInfo was package-private - let me know if I can be of any help.

**Comment by liutikas** [comment]
Any updates on this? It would be great to have a fix or a workaround for this.

**Comment by raphw** [comment]
I have attempted a fix. Could you try it out once it is merged and confirm that it works?

**Comment by szczepiq** [comment]
@liutikas, can you locally build mockito with PR: https://github.com/mockito/mockito/pull/1128 and let us know if it improves the behavior? @raphw, thank you very much for the fix!!!

**Comment by karczews** [comment]
I've stumble upon exactly the same problem when mocking android classes (dexmaker didnt have this pro blem). Fix from PR#1128 seems to solve the issue. Thanks! 
Can't wait for new release.

**Comment by TimvdLippe** [comment]
Closing per above comment. Feel free to reopen if you can still reproduce this with Mockito version 2.10.0 or later!
